File: backend/src/Job.java
/*
Class contains a constructor to create a job as well as
specific job attributes including:
-creator
-type: research, TA, other
-major(s)
-title
-description
-GPA requirement
-class year(s) requirement
 */

public class Job {
    int id; // unique id for this job
    String creator; // username of the employer who created the job
    int type; // 0=research, 1=TA, 2=other
    HashList<String> majors; // majors required for student
    String title; // name of the job position
    String department; // department name
    String location; // name of the location
    int numStudents; // number of students wanted to hire
    int hours; // hours per week
    String email; // contact email
    boolean federalWorkStudy; // if it counts for federal work study
    String desc; // description of the job position
    float gpa; // required GPA to apply
    int classYear; // required class year
    int grad; // 0 = undergrad, 1 = grad, 2 = postdoc
    String phone;
    String contact; // contact person

    public Job() {
//        creator = ;
//        type = ;
//        majors.add();
//        title = ;
//        desc = ;
//        gpa = ;
//        classYear = ;
//        email = ;

    }


}

File: hack-2023/src/jsonObjects/Job.ts
export class Job{
    public id: number;
    public creators: string[];
    public type: number;
    public majors: string[];
    public title: string;
    public department: string;
    public location: string;
    public numStudents: number;
    public hours: number;
    public email: string;
    public federalWorkStudy: boolean;
    public desc: string;
    public gpa: number;
    public classYear: number;
    public phone: string;
    public contact: string;
    public employees: string[];

    constructor(id: number, creators: string[], type: number, majors: string[], title: string, department: string, location: string, numStudents: number, hours: number, email: string, federalWorkStudy: boolean, desc: string, gpa: number, classYear: number, phone: string, contact: string, employees: string[]) {
        this.id = id;
        this.creators = creators;
        this.type = type;
        this.majors = majors;
        this.title = title;
        this.department = department;
        this.location = location;
        this.numStudents = numStudents;
        this.hours = hours;
        this.email = email;
        this.federalWorkStudy = federalWorkStudy;
        this.desc = desc;
        this.gpa = gpa;
        this.classYear = classYear;
        this.classYear = classYear;
        this.phone = phone;
        this.contact = contact;
        this.employees = employees;
    }

}